bigflow/dataflow.py

```diff
@@ -12,31 +13,35 @@
 
 class BeamJob(Job):
     def __init__(
-        self,
-        id: str,
-        entry_point: typing.Callable[[Pipeline, JobContext, dict], None],
-        pipeline_options: PipelineOptions,
-        entry_point_arguments: typing.Optional[dict] = None,
-        wait_until_finish: bool = True,
+            self,
+            id: str,
+            entry_point: typing.Callable[[Pipeline, JobContext, dict], None],
+            pipeline_options: PipelineOptions,
+            pipeline: Type[Pipeline],
```

Comment:
Assuming we want this parameter for tests:

1. It should be optional
2. It should be a pipeline object, not a pipeline class (so, for example, a user can use `with TestPipeline() as p` clause)
3. It should be called `test_pipeline`, to strongly suggest the purpose

Reply:
ad1. Changed, see now.
ad2. If we pass an instance of the Pipeline class then we will not be able to apply_logging inside the execute method. In that case, the user has to provide logging by himself.
ad3 I do not like `test_pipeline` name,  maybe I will add info about test purpose in docs?

Reply:
ad2. and thats ok, because it's for testing purposes
ad3. it's for testing, so why not `test_pipeline`?

Reply:
ad2. That is ok, but I'd add some checks like "assert not pipeline or not pipeline_options"
So user can specify either "pipeline_options" OR pipeline object.

ad3. I also don't like `test_pipeline` name, but It might protect our users from creating pipelines manually in the prod code, which is good. So I'd vote for "ugliest name" (eg `test_pipeline`).